src/web-eid.ts

```diff
@@ -20,51 +20,71 @@
  * SOFTWARE.
  */
 
-import config from "./config";
+import * as version from "./utils/version";
 
-import ErrorCode from "./errors/ErrorCode";
+import {
+  ExtensionAuthenticateRequest,
+  ExtensionGetSigningCertificateRequest,
+  ExtensionSignRequest,
+  ExtensionStatusRequest,
+} from "./models/message/ExtensionRequest";
+import {
+  ExtensionAuthenticateResponse,
+  ExtensionGetSigningCertificateResponse,
+  ExtensionSignResponse,
+  ExtensionStatusResponse,
+} from "./models/message/ExtensionResponse";
+import {
+  LibraryAuthenticateResponse,
+  LibraryGetSigningCertificateResponse,
+  LibrarySignResponse,
+  LibraryStatusResponse,
+} from "./models/message/LibraryResponse";
 
 import Action from "./models/Action";
-import AuthenticateOptions from "./models/AuthenticateOptions";
-import SignOptions from "./models/SignOptions";
-import Versions from "./models/Versions";
-import ResponseAuthenticateSuccess from "./models/ResponseAuthenticateSuccess";
-import ResponseStatusSuccess from "./models/ResponseStatusSuccess";
-
+import ActionOptions from "./models/ActionOptions";
+import ErrorCode from "./errors/ErrorCode";
+import MissingParameterError from "./errors/MissingParameterError";
+import VersionMismatchError from "./errors/VersionMismatchError";
 import WebExtensionService from "./services/WebExtensionService";
+import config from "./config";
+import sleep from "./utils/sleep";
 
-import * as version from "./utils/version";
-import VersionMismatchError from "./errors/VersionMismatchError";
-import HttpResponse from "./models/HttpResponse";
-import MissingParameterError from "./errors/MissingParameterError";
-import defer from "./utils/defer";
-import ResponseSignSuccess from "./models/ResponseSignSuccess";
 
 
 const webExtensionService = new WebExtensionService();
+const initializationTime  = +new Date();
 
-export async function status(): Promise<Versions> {
-  await defer(); // Give chrome a moment to load the extension content script
+/**
+ * Give browsers a moment to load the content script
+ */
+async function extensionLoadDelay(): Promise<void> {
+  const now = +new Date();
+  await sleep(initializationTime + config.MAX_EXTENSION_LOAD_DELAY - now);
```

Comment:
Gosh, this adds an unconditional delay... can we poll the extension until it responds instead (possibly instantly)?

Reply:
From my testing, there are several factors which influence how fast the extension's content script gets added to the page:
* Which browser is used
* Current CPU and/or IO usage
* Is the browser DevTools open
* Time from the Web-eID library loading to website's DOM being ready (blocking assets requests vs async)
* Extension code complexity (which we now reduced)

This makes reproducing the issue consistently a bit tricky.  
Currently the `MAX_EXTENSION_LOAD_DELAY` is `1000` ms. From my testing it was usually between 10-100ms, but when artificially stressing the PC and testing inside a VM with limited resources and having the DevTools open, up to 530ms.

It could be solved so that the `web-eid:status` message could be retried automatically a couple of times when `web-eid:status-ack` message isn't received from the extension.  

This is out-of-scope, should be planned separately.

Reply:
Alright, I created WE2-614 to track this.